### ISSUE TITLE
ComposeContentEmitterReturningValues incorrectly reports on compose functions with multiple top-level emitters

### ISSUE THREAD

**Issue by acrab** [slackhq/compose-lints#339]
Example code: 
```kotlin
@Composable
fun Test(modifier: Modifier = Modifier){
    Text(text = "TextOne")
    Text(text = "TextTwo")
}
```

Expected result: [ComposeMultipleContentEmitters](https://github.com/slackhq/compose-lints/blob/main/compose-lint-checks/src/main/java/slack/lint/compose/MultipleContentEmittersDetector.kt) lint rule will trigger

Actual result: Both that rule and [ComposeContentEmitterReturningValues](https://github.com/slackhq/compose-lints/blob/main/compose-lint-checks/src/main/java/slack/lint/compose/ContentEmitterReturningValuesDetector.kt) trigger, leading to a confusing set of warnings!

![image](https://github.com/slackhq/compose-lints/assets/7004090/a620af09-dc6d-4283-9383-4457b5f69c59)

**Comment by ZacSweers** [comment]
PR welcome!
